tests/vet/get-available-dogs.test.ts

```diff
@@ -0,0 +1,176 @@
+import { getAvailableDogs } from "@/lib/vet/actions/get-available-dogs";
+import { withDb } from "../_db_helpers";
+import {
+  getDogOii,
+  getEligibleDogSpecOverrides,
+  getVetActor,
+  insertCall,
+  insertDog,
+  insertUser,
+  insertVet,
+} from "../_fixtures";
+import { Pool } from "pg";
+import {
+  dbInsertDogVetPreference,
+  dbUpdateDogParticipation,
+} from "@/lib/data/db-dogs";
+import {
+  CALL_OUTCOME,
+  PARTICIPATION_STATUS,
+  USER_RESIDENCY,
+  YES_NO_UNKNOWN,
+} from "@/lib/data/db-enums";
+import { AvailableDog } from "@/lib/vet/vet-models";
+import { dbQuery } from "@/lib/data/db-utils";
+
+describe("getAvailableDogs", () => {
```

Comment:
Also add this test

- it should not return available dog of another vet

Reply:
Added the test in https://github.com/barkbank/barkbank-pawtal/pull/114/commits/faacaed656d8003cd8ed4680ecc5a9da55c0df09